### ISSUE TITLE
bug: WCAG 2.2 guideline 1.3.1 failed: button property in card makes cards not comply with WCAG

### ISSUE THREAD

**Issue by ijsje71** [ionic-team/ionic-framework#30407]
### Prerequisites

- [x] I have read the [Contributing Guidelines](https://github.com/ionic-team/ionic-framework/blob/main/docs/CONTRIBUTING.md#creating-an-issue).
- [x] I agree to follow the [Code of Conduct](https://ionicframework.com/code-of-conduct).
- [x] I have searched for [existing issues](https://github.com/ionic-team/ionic-framework/issues) that already report this problem, without success.

### Ionic Framework Version

v8.x

### Current Behavior

When the button property is added to a card, the card styling is rendered in <button> tags. This violates WCAG guideline 1.3.1 and is hard for a screenreader to comprehend, as there is a lot of styling within those cards. A button normally does not have titles, subtitles etc. inside, which is functionality that is available for cards.

When the button is removed, the card is still clickable, but screen readers then do not know that this is a clickable item.

### Expected Behavior

When the card is tappable, it is rendered through links or another interactive component. This should not violate WCAG guidelines.

We currently have a workaround that uses href="javascript:void(0)" to invoke the onclick method for screen readers as well, but I would prefer to see an out-of-the-box solution to tappable buttons that is WCAG compliant.

### Steps to Reproduce

1. Go to any card implementation that includes the button property
2. Go to the shadow DOM
3. Notice the button tags around the card
4. Now go to https://www.w3.org/WAI/WCAG22/Understanding/info-and-relationships.html
5. Notice the difference between the guidelines and the button tag in the card implementation

### Code Reproduction URL

I could make a git repo from this, but seems a bit overkill for just a card...

### Ionic Info

Ionic:

   Ionic CLI       : 7.2.0
   Ionic Framework : @ionic/react 8.5.7

Capacitor:

   Capacitor CLI      : 7.2.0
   @capacitor/android : not installed
   @capacitor/core    : 7.2.0
   @capacitor/ios     : not installed

Utility:

   cordova-res : not installed globally
   native-run  : 2.0.1

System:

   NodeJS : v22.14.0
   npm    : 10.9.2
   OS     : macOS Unknown

**Comment by ijsje71** [comment]
@thetaPC do you have any teams that are working on the accessibility issues? A lot has been done already, but issues like this are forcing hacky workarounds to comply with WCAG standards

**Comment by thetaPC** [comment]
Thank you for submitting the issue!

I read through the provided link and was unable to find the connection between buttons and cards. Please provide any information that clearly shows the violation. Examples would be great!
